lib/AST/Requirement.cpp

```diff
@@ -332,4 +332,29 @@ void InverseRequirement::expandDefaults(
                          SourceLoc()});
     }
   }
-}
\ No newline at end of file
+}
+
+/// Determines the canonical ordering of inverse requirements for a subject.
+/// NOTE: this order impacts ABI stability!
+static unsigned getInvertibleProtocolOrder(InvertibleProtocolKind kind) {
```

Comment:
Why don't you use the protocol order that's already used for generic signatures (`TypeDecl::compare()`)

Reply:
yeah I don't know how I missed that :)